core/src/main/scala/dx/core/ir/ValueSerde.scala

```diff
@@ -430,10 +429,9 @@ object ValueSerde extends DefaultJsonProtocol {
           // in fieldTypes that do not appear in fields are optional
           val keys1 = fields.keySet
           val keys2 = fieldTypes.keySet
-          val extra = keys2.diff(keys1)
-          if (extra.nonEmpty) {
+          if (!keys1.subsetOf(keys2)) {
             throw ValueSerdeException(
-                s"struct ${name} value has members that do not appear in the struct definition: ${extra}"
+                s"struct definition: ${keys1}, has members that do not appear in struct ${name}"
```

Comment:
This message is wrong. `keys1` are the keys in the struct value.

Reply:
Check if the current error message is correct, feel free to update it if needed.